db/lock.go

```diff
@@ -58,21 +57,23 @@ func setDocumentLocked(id string, upsert bool) (bool, error) {
 	// for findAndModify-ing the lock
 
 	// timeout to check for
-	timeoutThreshold := time.Now().Add(-LockTimeout)
+	timeoutThreshold := time.Now().Add(-lockTimeout)
```

Comment:
Why not `time.Now().Sub(lockTimeout)`?

Reply:
.Sub() takes a time.Time while .Add() takes a time.Duration, so if you want to subtract a duration from a time, you have to use .Add() with a negative duration.